ibis/backends/snowflake/__init__.py

```diff
@@ -406,20 +406,22 @@ def _register_in_memory_table(self, op: ops.InMemoryTable) -> None:
                 )
                 for col, typ in op.schema.items()
             )
-            con.exec_driver_sql(f"CREATE TEMP TABLE {table} ({schema})")
-
-            # 4. copy the data into the table
-            columns = op.schema.names
-            column_names = ", ".join(map(quote, columns))
-            parquet_column_names = ", ".join(f"$1:{col}" for col in columns)
-            con.exec_driver_sql(
-                f"""
-                COPY INTO {table} ({column_names})
-                FROM (SELECT {parquet_column_names} FROM @{stage})
-                FILE_FORMAT=(TYPE=PARQUET COMPRESSION=AUTO)
-                PURGE=TRUE
-                """
-            )
+            if con.exec_driver_sql(f"SHOW TABLES LIKE '{table}'").scalar() is None:
+                con.exec_driver_sql(
+                    f"CREATE TEMP TABLE IF NOT EXISTS {table} ({schema})"
```

Comment:
IIUC the `if` statement above is checking if the table already exists before running this block - if so, should we drop the `"IF NOT EXISTS"` here then?

Reply:
Yeah, thanks for catching!

Reply:
Done! Added a test too!